sharding/shards_ring.go

```diff
@@ -133,8 +139,8 @@ func (sr *ShardsRing) logInconsistency(key, expectedClusterName, actualClusterNa
 	}
 }
 
-// RoundTrip performs http requests
-func (sr ShardsRing) RoundTrip(req *http.Request) (resp *http.Response, rerr error) {
+//DoRequest performs http requests
```

Comment:
What it does? What it returns and when? How it handle nil? Compare with https://golang.org/pkg/net/http/#Client.Do